services/order.py

```diff
@@ -0,0 +1,37 @@
+from django.db import transaction, IntegrityError
+from db.models import Order, Ticket, User, MovieSession
+
+
+@transaction.atomic
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str | None = None
+) -> list[dict]:
+    try:
+        user = User.objects.get(username=username)
+        order = Order.objects.create(user=user)
+        if date:
+            order.created_at = date
+            order.save()
+
+        for ticket_data in tickets:
+            movie_session = MovieSession.objects.get(
+                id=ticket_data["movie_session"]
+            )
+            Ticket.objects.create(
+                movie_session=movie_session,
+                order=order,
+                row=ticket_data["row"],
+                seat=ticket_data["seat"]
+            )
+        return order
+    except IntegrityError:
```

Comment:
no need to add this try-except, transaction mechanism will do a rollback - it is its main purpose